ipi/engine/forcefields.py

```diff
@@ -375,6 +373,97 @@ def evaluate(self, r):
         r["status"] = "Done"
 
 
+try:
+    import quippy
+except Exception as e:
+    quippy = None
+    quippy_exc = e
+
+
+class FFQUIP(ForceField):
+
+    """Basic fully pythonic force provider.
+
+    Computes an arbitrary interaction potential implemented in QUIP. 
+    Parallel evaluation with threads.
+
+    Attributes:
+        parameters: A dictionary of the parameters used by QUIP. Of the
+            form {'name': value}.
+        requests: During the force calculation step this holds a dictionary
+            containing the relevant data for determining the progress of the step.
+            Of the form {'atoms': atoms, 'cell': cell, 'pars': parameters,
+                         'status': status, 'result': result, 'id': bead id,
+                         'start': starting time}.
+    """
+
+    def __init__(self, init_file, args_str, param_file, latency=1.0e-3, name="", pars=None, dopbc=True):
+        """Initialises QUIP.
+
+        Args:
+        pars: Mandatory dictionaru, giving the parameters needed by QUIP.
+        """
+        if quippy is None:
+            raise ImportError("QUIPPY import failed due to exception : " + str(e))
```

Comment:
I would also consider doing something other than `ImportError` here. This is the error the user will see they don't have quippy installed, so ideally it would not end up as an unhandled exception. Print error message and exit with a non-zero return code would be the expected behaviour in a case like this, I think.